drop-transfer/src/error.rs

```diff
@@ -74,51 +74,53 @@ impl Error {
     }
 }
 
-impl From<&Error> for u32 {
+impl From<&Error> for drop_core::Status {
     fn from(err: &Error) -> Self {
         use drop_core::Status;
 
         match err {
-            Error::Canceled => Status::Canceled as _,
-            Error::BadPath(_) => Status::BadPath as _,
-            Error::BadFile => Status::BadFile as _,
-            Error::BadTransfer => Status::BadTransfer as _,
-            Error::BadTransferState(_) => Status::BadTransferState as _,
-            Error::BadFileId => Status::BadFileId as _,
-            Error::Io(io) => {
-                let status = match io.kind() {
-                    ErrorKind::PermissionDenied => Status::PermissionDenied,
-                    _ => Status::IoError,
-                };
-
-                status as _
-            }
-            Error::DirectoryNotExpected => Status::BadFile as _,
-            Error::TransferLimitsExceeded => Status::TransferLimitsExceeded as _,
-            Error::MismatchedSize => Status::MismatchedSize as _,
-            Error::UnexpectedData => Status::MismatchedSize as _,
-            Error::InvalidArgument => Status::InvalidArgument as _,
-            Error::WsServer(_) => Status::IoError as _,
-            Error::WsClient(_) => Status::IoError as _,
-            Error::AddrInUse => Status::AddrInUse as _,
-            Error::FileModified => Status::FileModified as _,
-            Error::FilenameTooLong => Status::FilenameTooLong as _,
-            Error::AuthenticationFailed => Status::AuthenticationFailed as _,
-            Error::StorageError(_) => Status::StorageError as _,
-            Error::ChecksumMismatch => Status::FileChecksumMismatch as _,
-            Error::FileStateMismatch(FileTerminalState::Rejected) => Status::FileRejected as _,
-            Error::FileStateMismatch(FileTerminalState::Completed) => Status::FileFinished as _,
-            Error::FileStateMismatch(FileTerminalState::Failed) => Status::FileFailed as _,
-            Error::EmptyTransfer => Status::EmptyTransfer as _,
-            Error::ConnectionClosedByPeer => Status::ConnectionClosedByPeer as _,
-            Error::TooManyRequests => Status::TooManyRequests as _,
+            Error::Canceled => Status::Finalized,
```

Comment:
Let's rename `Error::Canceled` -> `Finalized` as well

Reply:
Huh, now as I think of it, it doesn't make sense for the error, because it means that something is canceled. For example, this error is handled cleanly (without emitting an event) in many cases.
So I see this as a different entity from `Finalization` and I think we should leave it as it is

Reply:
Thanks for explaining. Where is it used though? In a context of transfer or a file?

Reply:
In the context of file, for example, it's used to signal disconnection from the WS layer to the logic above (`Paused` event emitted)

Reply:
Thanks for explaining, now it's clear. It feels like this enum value should hold more information why it was cancelled or else we're just guessing, wonder if that doesn't hide any bugs.. It was always there, just is now highlighted due to a rename.

By the way
```
...
elif ev.is_transfer_failed():
        return event.FinishFailedTransfer(
            transfer_slot, ev.status.status, ev.status.os_error_code
        )
    elif ev.is_transfer_deferred():
        return event.TransferDeferred(
            transfer_slot, ev.peer, ev.status.status, ev.status.os_error_code
        )
...
```

These functions are generated by UniFFI I assume? How do they work, does UniFFI generate `is_*` itself or do we have something tied to it somewhere(can;t find it)?

Reply:
That's an additional point for commiting generated code